### PR TITLE
feature(docs): modal component theming

### LINKED ISSUE #1053: Theming: Modal
### Subject

Modal

### Description

Create a basic theming documentation for the `Modal` component that showcases how you can style it.

Theming file: [Modal theming file](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/modal/theming.mdx)

### Template Reference

For single part components, feel free to use the `Button` component theming doc as a template reference: [Button theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/button/theming.mdx).

For multipart components, feel free to use the `Input` component theming doc as a template reference: [Input theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/input/theming.mdx).

### PR DESCRIPTION
**Pull request by lejara**
Closes #1053

## 📝 Description

Adds theming  docs to the modal component with an example.

I noticed there are updates to other theming docs. So I tired to match its structure accordingly 

## 💣 Is this a breaking change:

No

### CODE REVIEW

File: configs/sandpack-contents/component-theming/modal.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  
+  export const modalTheme = defineMultiStyleConfig({
+    baseStyle,
+    defaultProps: {
+      colorScheme: "purple", //set the default color scheme to purple
+      size: "2xl",
+    },
+  });
+  
+  `,
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/modal.js:112]: Extra lines at the top and bottom + extra indentation.

File: content/docs/components/modal/theming.mdx [hunk 2]
```diff
@@ -2,3 +2,49 @@
...
+  ModalTheme,
+} from 'configs/sandpack-contents/component-theming/modal.js'
+
+<SandpackEmbed
+  files={{
+    '/theme/components/modal.ts': ModalTheme,
+    '/App.tsx': App,
+    '/index.tsx': {
+      code: Index,
+      hidden: false,
```
Comment (nikolovlazar) [inline content/docs/components/modal/theming.mdx:44]: Why show this file? 😅

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
Hello @lejara. I've left you some comments on indentation and extra lines. I'll do another review tomorrow on the example itself. Thanks for working on this 🙌

File: content/docs/components/modal/theming.mdx (hunk 2 above)
Comment (lejara) [reply content/docs/components/modal/theming.mdx:44]: I personally think its a quality of life improvement. It shows how you extend the theme with said component without going to another page in the docs.

Though i get how this is debatable. So ill just hide it again.

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
@lejara we've updated the multipart component theming doc template to provide more value to the users by including more use cases and guides. Please check out the new structure and optimize your PR according to it.

**Comment by lejara** [comment]
sure thing!

File: content/docs/components/modal/theming.mdx [hunk 3]
```diff
@@ -2,3 +2,198 @@
...
+- `footer`
+
+You can find more information in the source
+[here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/modal.ts).
+
+## Theming properties
+
+The properties that affect the theming of the `Modal` component are:
+
+- `baseStyle`: The base style of `Modal`.
```
Comment (nikolovlazar) [inline content/docs/components/modal/theming.mdx:31]: We shouldn't add `baseStyle` here. It's not a theming property.

File: content/docs/components/modal/theming.mdx [hunk 4]
```diff
@@ -2,3 +2,198 @@
...
+  baseStyle,
+})
+```
+
+After customizing the modal theme, we can import it in our theme file and add it
+in the components property:
+
+```jsx live=false
+import { extendTheme } from '@chakra-ui/react'
+import { modalTheme } from './components/theme/modal.ts'
```
Comment (nikolovlazar) [inline content/docs/components/modal/theming.mdx:74]: Remove the `.ts` at the end of the import.

File: content/docs/components/modal/theming.mdx [hunk 5]
```diff
@@ -2,3 +2,198 @@
...
+  sizes,
+})
+
+// Now we can use the new `xl` size
+<Modal size="xl" ... />
+```
+
+> Every time you're adding anything new to the theme, you'd need to run the CLI
+> command to get proper autocomplete in your IDE. You can learn more about the
+> CLI tool [here](/docs/styled-system/cli).
```
Comment (nikolovlazar) [inline content/docs/components/modal/theming.mdx:123]: This doesn't need to be a callout. Let's remove the `>`.

File: configs/sandpack-contents/component-theming/modal.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  Button,
+  Box,
+  SimpleGrid,
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/modal.js:13]: We're not using the `SimpleGrid`, so we can remove it.

File: configs/sandpack-contents/component-theming/modal.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+          <ModalFooter>
+            <Button colorScheme="blue" mr={3} onClick={onClose}>
+              Close
+            </Button>
+            <Button variant="ghost">Secondary Action</Button>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
+      <IconButton
+        rounded="full"
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/modal.js:51]: We should add a `aria-label="change theme"` to fix the issue shown in CodeSandbox.

File: configs/sandpack-contents/component-theming/modal.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      />
+    </Box>
+  );
+}`,
+  Index: `import * as React from "react";
+import { createRoot } from "react-dom/client";
+import { ChakraProvider, extendTheme } from "@chakra-ui/react";
+
+import App from "./App";
+import { modalTheme } from "./theme/components/modal.ts";
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/modal.js:67]: Remove the `.ts` from the import.

File: content/docs/components/modal/theming.mdx [hunk 9]
```diff
@@ -2,3 +2,198 @@
...
+- `header`
+- `overlay`
+- `dialogContainer`
+- `dialog`
+- `closeButton`
+- `body`
+- `footer`
+
+You can find more information in the source
+[here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/modal.ts).
```
Comment (nikolovlazar) [inline content/docs/components/modal/theming.mdx:25]: I've prepared an illustration for you. Put the `AspectRatio` below the anatomy list (see the Input or Menu theming docs to see exactly here). Here's the code for the illustration: https://gist.github.com/nikolovlazar/bb69fcd8d035bad0adbf372e033f86ff

Also add letters to the parts:
```
- A: `header`
- B: `overlay`
- C: `dialogContainer`
- D: `dialog`
- E: `closeButton`
- F: `body`
- G: `footer`
```

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
Just a few more changes and we're good!

**Review by nikolovlazar — APPROVED** [review]
All good! Thanks for contributing to the Modal theming @lejara!

**Comment by nikolovlazar** [comment]
@allcontributors please add @lejara for docs.

**Pull request merged**